server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -793,6 +817,15 @@ void handleSimpleQuery(ByteBuf buffer, final DelayableWriteChannel channel) {
         }
     }
 
+    private void handleCancelRequestBody(ByteBuf buffer, Channel channel) {
+        var keyData = readCancelRequestBody(buffer);
+        Session s = ACTIVE_SESSIONS.get(keyData); // the session executing the target query
+        if (s == null) {
+            return;
+        }
+        tasksService.killJobs(List.of(s.getActiveJobID()), "crate", null);
```

Comment:
This would only kill the job on the local node, but not on other nodes.
The kill instruction needs to broadcast to all nodes in the cluster. See `KillPlan.execute` for an example